core/src/test/java/jenkins/util/io/PathRemoverTest.java

```diff
@@ -51,10 +51,16 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
 
+import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.arrayWithSize;
 import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasItem;
-import static org.junit.Assert.*;
```

Comment:
Now that `org.junit.Assert.assertThat` is deprecated, we have to import all of the methods individually so that we can import and use `org.hamcrest.MatcherAssert.assertThat` to avoid deprecation warnings without having to write that out fully-qualified every time.

Reply:
Wildcard import of `org.hamcrest.MatcherAssert.*` has been the preferred technique for me thus far.  That prepares for JUnit 5 (a long ways away for us because we need `JenkinsRule`).

Can you help me understand why the wildcard import of `org.hamcrest.MatcherAssert.*` won't work in this case?

Reply:
It looks like `MatcherAssert`  only provides `assertThat`, but existing tests in this file are using `assertTrue`, `assertFalse`, etc. from `org.junit.Assert`, so we still need to pull in those methods somehow. `org.junit.Assert` also provides `assertThat`, but that method is deprecated, so we can't do a wildcard import of `org.junit.Assert.*` _if_ we want to use `assertThat` from `MatcherAssert` without having to fully-qualify it at ever call site.

We could convert all of the tests in this file to use `assertThat` so we could just import `org.hamcrest.MatcherAssert.*` and not anything in the `org.junit.Assert` package, but that seems like something best left for a different PR.

Reply:
Very good point.  I've used the deprecation of org.junit.Assert assertions as a good excuse to replace assertTrue and assertFalse with assertThat.  The resulting failure messages tend to be better with assertThat than they were with assertFalse and assertTrue..

Agreed wholeheartedly that changing the assertions in tests is for a separate PR, unrelated to this PR.